### DISCUSSION TITLE
Request for Kagikn - Bypass Update Check

### DISCUSSION THREAD

**Discussion by JohnFromGWN** [scripthookvdotnet/scripthookvdotnet#1540] [General]
I know you probably have much better things to do with your time, but I'll ask anyway on behalf of the reverted game community. Back in April of 2023 you saved all the "reverters" with your Integrity Bypass Against RGL which was really about the RGSC.

https://github.com/kagikn/ExeIntegrityBypassAgainstRGL

https://www.gta5-mods.com/tools/exe-integrity-bypass-against-rgl

Since 3411, R* has been forcing updates with a different methodology and we can no longer revert without a ridiculous process of going offline, being forced to authenticate, faking an update, launching game online, reverting, and finally going back once more offline. 

I assume creating a new enabler to bypass this new R* police tactic would be quite an undertaking, but what the hell, I'm going to ask anyway. In passing, most users don't realize that the reverting process only requires one file, and that's GTA5.exe. I created a batch for myself, which i'll share below, which facilitates the process, but of course you can't bypass (without a mod) having to launch the game to authenticate.

This allows you to swap exe's in or out. Obviously you need to adjust the path and you can rename the files whatever you want.
You will also need to remove -scOfflineOnly from the Steam launch options and instead start the game like this (you can also put in your steam password and id from the command line):
`start "" "C:\Program Files (x86)\Steam\steam.exe" -applaunch 271590 -scOfflineOnly`

```
@echo off
cd /d "P:\SteamLibrary\steamapps\common\Grand Theft Auto V"

REM This is to revert to 2372
if exist "GTA5_2372.exe" (
    ren "GTA5.exe" "GTA5_Current.exe"
    ren "GTA5_2372.exe" "GTA5.exe"
	ren "dinput8.bak" "dinput8.dll"
    echo Renamed GTA5.exe to GTA5_Current.exe and GTA5_2372.exe to GTA5.exe

start "" "C:\Program Files (x86)\Steam\steam.exe" -applaunch 271590 -scOfflineOnly
exit
) else (
    REM Check if GTA5_Current.exe exists
    if exist "GTA5_Current.exe" (
        ren "GTA5.exe" "GTA5_2372.exe"
        ren "GTA5_Current.exe" "GTA5.exe"
		ren "dinput8.dll" "dinput8.bak"
        echo Renamed GTA5.exe to GTA5_2372.exe and GTA5_Current.exe to GTA5.exe
@echo off
start "" "C:\Program Files (x86)\Steam\steam.exe" -applaunch 271590
exit
    ) else (
        echo Neither GTA5_2372.exe nor GTA5_Current.exe exist. No renaming performed.
    )
)
```

**Comment by JohnFromGWN** [comment]
Note: renaming dinput8.dll is optional but is there to cover the cases when Mr. Blade has not yet released a current version of SH5, resulting in a version error. I need this myself because I don't bother to update SH5 - both my exe and SH5 are frozen as 2372.

Removing -scOfflineOnly from the Steam Launch Options for GTA V is also optional but if you don't you will have to manually remove it each time you're asked to authenticate, and that will happen every 3 to 5 days.

Finally, sure you can used cracked dlls as a bypass but you're exposing yourself to malware that is undetectable by any anti-virus software.

**Comment by kagikn** [comment]
Definitely not within the organization's scope, and actually not within the scope of that tool of mine. You'll need to add the `nobattleye` launcher option or use a newer SHV, and the game also requires a corresponding `title.rgl` on all platforms since b3323 (tied to GTA5.exe), but I'm not discussing the topic further.

**Reply by JohnFromGWN** [reply]
Not a problem. I expected a negative answer and thankfully I created a workaround so I don't depend on others.  In passing, this issue has absolutely nothing to do with battleye which is only pertinent to GTAO and anti-cheating. Also it has nothing to do with SHV or SHVDN, neither of which will stop a R* update. 

This was purely about bypassing a forced update. Just a final word, it's curious why your RGSC was within scope but this isn't. Have a wonderful new year and I will not bother you again.

**Reply by kagikn** [reply]
The issuer test issue against `socialclub.dll` is done by GTA5.exe until b2824 stopped doing, which ASI modules can interfere. On the other hand, matching title.rgl test isn't done by GTA5.exe but is done by some RGSC module, which ASI modules in the game directory can't interfere. I wasn't sure until now, but now I'm sure the test isn't done by GTA5.exe because you can't even see a process of GTA5.exe with Task Manager or a similar tool when you see "Update Required" by not matching title.rgl.

But again, this discussion isn't supposed to belong here but could *have* posted in the issue of [my ExeIntegrityBypassAgainstRGL repo](https://github.com/kagikn/ExeIntegrityBypassAgainstRGL/issues) if the title.rgl test *was* done by GTA5.exe, though it's inappropriate because GTA5.exe doesn't test that.

EDIT: for the Steam game versions that do not have title.rgl on Steam (e.g. v1.0.2060.0), I can play the game with a `title.rgl` for b3351 or older. ~I can play the game with corresponding `PlayGTAV.exe` and `GTAVLauncher.exe` .~

**Reply by JohnFromGWN** [reply]
Hi
I had written to you here because I didn't know any other way to contact you and had downloaded your previous bypass mod from GTA5Mods, not even thinking of Github. I'll repost this topic on your link. Thank you.

This is what I can tell you regarding the version test from my perspective.  My Steam game is 100% updated except for 1 file: GTA5.exe. From that we have to conclude that whatever test is done, the target is GTA5.exe. We can rule out any "flag" in the Windows registry or buried somewhere else. Similarly, since I can swap my GTA5.exe for an older copy and go offline, we can rule out all other GTA5 files since mine are all updated to the current version. If I go online with the old GTA5.exe, in that case an error msg will appear telling me to update. If then I restart the game offline there won't be any error message. 

The mechanism, I'm using now, using  a cmd file works but requires being offline. This is a minor issue - but by being offline, with SP,  you will lose functionality and ofc the nag to authenticate after a few days - that requires swapping back the current exe.

A bigger concern, regarding these new forced updates, is that R* could eventually create a RGL or RGSC change that eliminates modding completely in a shotgun approach to end cheating in GTAO.
